packages/cli/src/ErrorHandling.ts

```diff
@@ -0,0 +1,47 @@
+import * as Sentry from '@sentry/node';
+import { RewriteFrames } from '@sentry/integrations';
+import type { Application } from 'express';
+import config from '../config';
+import { ErrorHandler } from 'n8n-workflow';
+
+let initialized = false;
+
+export const initErrorHandling = (app?: Application) => {
+	if (initialized) return;
+
+	if (!config.getEnv('diagnostics.enabled')) {
+		initialized = true;
+		return;
+	}
+
+	const dsn = config.getEnv('diagnostics.config.sentry.dsn');
+	const { N8N_VERSION: release, ENVIRONMENT: environment } = process.env;
+
+	Sentry.init({
+		dsn,
+		release,
+		environment,
+		integrations: (integrations) => {
+			integrations.push(new RewriteFrames({ root: process.cwd() }));
+			return integrations;
+		},
+	});
+
+	if (app) {
+		const { requestHandler, errorHandler } = Sentry.Handlers;
+		app.use(requestHandler());
+		app.use(errorHandler());
+	}
+
+	initialized = true;
+};
+
+export const captureError = Sentry.captureException;
+
+export const captureWarning = (warning: string) =>
+	Sentry.captureMessage(warning, { level: 'warning' });
+
+Object.assign(ErrorHandler, {
+	captureError,
+	captureWarning,
+});
```

Comment:
The `init` here makes me think this could have been a class with a constructor. Instead of functions in a module, `Logger`, `WaitTracker`, `WorkflowRunner`, etc. are all classes. Are we consciously moving away from this style?

Reply:
We have a few "singleton" patterns. It's hard to decide which one to go with. This is following the pattern `Db.ts`. I can change this to be a singleton class.

Reply:
I've created a new `ErrorReporterProxy` like `LoggerProxy`. Can you please have a look again?